src/matrix/room/BaseRoom.js

```diff
@@ -415,7 +415,7 @@ export class BaseRoom extends EventEmitter {
     }
 
     get type() {
-        return this._summary.data.type;
+        return this._summary.data.type ?? undefined;
```

Comment:
what's this for?

Reply:
Fallback to `this._summary.data.type` is null when [processingStateEvent](https://github.com/vector-im/hydrogen-web/blob/bebbbd9db027ba8200e1f20be237dc63659de486/src/matrix/room/RoomSummary.js#L85) but it was returning undefined because of [this](https://github.com/vector-im/hydrogen-web/blob/bebbbd9db027ba8200e1f20be237dc63659de486/src/matrix/room/RoomSummary.js#L171) so added the ?? undefined to make sure it always return string or undefined at public endpoint.